pkg/relayer/config/supplier_hydrator.go

```diff
@@ -18,40 +23,116 @@ func (supplierConfig *RelayMinerSupplierConfig) HydrateSupplier(
 	// `HydrateSuppliers` is a part of `pkg/relayer/config/suppliers_config_hydrator.go`.
 	supplierConfig.SigningKeyNames = yamlSupplierConfig.SigningKeyNames
 
-	backendUrl, err := url.Parse(yamlSupplierConfig.ServiceConfig.BackendUrl)
+	// Hydrate the default service config
+	defaultServiceConfig, err := supplierConfig.hydrateServiceConfig(yamlSupplierConfig.ServiceConfig)
+	if err != nil {
+		return err
+	}
+	supplierConfig.ServiceConfig = defaultServiceConfig
+
+	// Hydrate the RPC type-specific service configs (if any)
+	supplierConfig.RPCTypeServiceConfigs = make(
+		map[types.RPCType]*RelayMinerSupplierServiceConfig,
+		len(yamlSupplierConfig.RPCTypeServiceConfigs),
+	)
+
+	// Loop through the RPC type-specific service configs and hydrate them.
+	// For example, if the supplier is configured to handle REST and JSON-RPC,
+	// there will be two RPC type-specific service configs.
+	for rpcType, serviceConfig := range yamlSupplierConfig.RPCTypeServiceConfigs {
+		rpcType, err := getRPCTypeFromConfig(rpcType)
+		if err != nil {
+			return err
+		}
+
+		rpcTypeServiceConfig, err := supplierConfig.hydrateServiceConfig(serviceConfig)
+		if err != nil {
+			return err
+		}
+
+		supplierConfig.RPCTypeServiceConfigs[rpcType] = rpcTypeServiceConfig
```

Comment:
Please add some debug logging on what is being configured when the RelayMiner starts.

Make it useful. Have emojis.

Everyone in the ecosystem looks at them@

Reply:
Done, I've added better logging.